src/sqlfluff/core/rules/__init__.py

```diff
@@ -4,17 +4,21 @@
 from sqlfluff.core.plugin.host import get_plugin_manager
 from sqlfluff.core.rules.config_info import get_config_info
 
-std_rule_set = RuleSet(name="standard", config_info=get_config_info())
-
-# Iterate through the rules list and register each rule with the std_rule_set
-plugin_manager = get_plugin_manager()
```

Comment:
Replaced the `import-time` rule discovery with deferred discovery. Otherwise, the plugin implementations have not yet been "discovered" at the time `std_rule_set` is being created and populated.